engine/access/state_stream/backend/backend_executiondata_test.go

```diff
@@ -38,12 +39,13 @@ var testEventTypes = []flow.EventType{
 type BackendExecutionDataSuite struct {
 	suite.Suite
 
-	state    *protocolmock.State
-	params   *protocolmock.Params
-	snapshot *protocolmock.Snapshot
-	headers  *storagemock.Headers
-	seals    *storagemock.Seals
-	results  *storagemock.ExecutionResults
+	state          *protocolmock.State
+	params         *protocolmock.Params
+	snapshot       *protocolmock.Snapshot
+	headers        *storagemock.Headers
+	seals          *storagemock.Seals
+	results        *storagemock.ExecutionResults
+	registersAsync *execution.RegistersAsyncStore
```

Comment:
I don't have much context on this but I wonder if the `Async` in the name makes sense? From just looking at it on the surface it feels like it's an implementation detail that doesn't need to be exposed, what is the asynchronicity that the consumer needs to be aware of?